03-git-github-and-markdown/COURSE_MATERIAL.md

```diff
@@ -817,55 +1012,201 @@ profile README!
+git status
+```
+
+The output should be similar to this:
+
+```text
+On branch main
+Your branch is up to date with 'origin/main'.
+
+Changes not staged for commit:
+  (use "git add <file>..." to update what will be committed)
+  (use "git restore <file>..." to discard changes in working directory)
+        modified:   README.md
+
+no changes added to commit (use "git add" and/or "git commit -a")
+```
+
+Check the differences between the working directory and the staging area.
+
+```sh
+# View the differences between the working directory and the staging area
+git diff README.md
+```
+
+The output should be similar to this:
+
+```diff
+diff --git a/README.md b/README.md
+index 7c46677..76e0528 100644
+--- a/README.md
++++ b/README.md
+@@ -1,2 +1,3 @@
+-# new-repository-demo
+-A repository to demonstrate the creation of a new repository.
++# How to create a new repository - demonstration
++
++A new line to my README file!
+```
+
+The `+` indicates the new line added to the `README.md` file.
+
+The `-` indicates the line removed from the `README.md` file.
+
+Add the `README.md` file to the staging area.:
 
 ```sh
 # Add the staged files to the staging area
 git add README.md
+```
 
-# Commit the changes
-git commit -m "Update profile README"
+Check the staging area.
+
+```sh
+# Check the staging area
+git status
+```
+
+The output should be similar to this:
+
+```text
+On branch main
+Your branch is up to date with 'origin/main'.
+
+Changes to be committed:
+  (use "git restore --staged <file>..." to unstage)
+        modified:   README.md
 ```
 
-> [!TIP]  
+If multiple files were modified, you would only see the files that were added to
+the staging area in the list of changes to be committed.
+
+Commit the changes with an appropriate message.
+
+> [!TIP]
+>
 > There is not a clear convention on how to write commit messages. The most
 > important thing is to be consistent. We have found using an infinitive verb
 > followed by a short description of the changes to be a good practice.
 
-#### Push your changes
+```sh
+# Commit the changes
+git commit -m "Update profile README"
+```
 
-Push your changes to the repository:
+Check the log to see the new commit.
 
 ```sh
-# Push the changes to the repository
-git push
+# View the commit history
+git log
 ```
 
-> [!NOTE]  
+The output should be similar to this:
+
+```text
+commit 2ea9f1c4d5ef42ee3cf1b0d3849a0445ad8d2683 (HEAD -> main)
+Author: Ludovic Delafontaine <ludovic.delafontaine@gmail.com>
+Date:   Tue Jul 23 18:33:12 2024 +0200
+
+    Demonstration of changes made to the new repository
+
+commit 20f87413aef2fe9ef296767e6db839b722b83532 (origin/main, origin/HEAD)
+Author: Ludovic Delafontaine <ludovic.delafontaine@gmail.com>
+Date:   Tue Jul 23 17:28:45 2024 +0200
+
+    Initial commit
+
+```
+
+You can notice that the new commit `2ea9f1c4` is the HEAD of the local `main`
+branch.
+
+The `origin/main` and `origin/HEAD` are still pointing to the `20f87413` commit
+as their HEAD.
```

Comment:
Is it normal not to see the latest “Update profile README” commit in the `git log`

Reply:
Fixed!